mantidproject/mantid
# Replace boost::shared_ptr with std::shared_ptr

**Description of work.**
This PR replaces the use of `boost::shared_ptr` with `std::shared_ptr` and extends to other functions such as `boost::make_shared`, `boost::weak_ptr`, dynamic pointer casts and `boost::atomic`.

**To test:**
Check Mantid builds successfully, tests pass and the GUIs function normally.

Closes #25842 

*This does not require release notes* because it is an internal change related to code quality**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
